### PR TITLE
Submit interactive jobs from remote host

### PR DESCRIPTION
**Pull request by sujatapatnaik52**
#### Describe Bug or Feature
PTL submits interactive jobs from server even though the client is a remote host.

#### Describe Your Change
Updated the interactive job class to ssh to client machines and submit as test user from the client. 

#### Attach Test Logs/Output
Few interactive job tests ran with -p client="hostname"
[smoketest.txt](https://github.com/openpbs/openpbs/files/5016167/smoketest.txt)
[suspend_resume_accounting.txt](https://github.com/openpbs/openpbs/files/5016172/suspend_resume_accounting.txt)

Regression results:
**Description**: Tests from given sources on platforms CENTOS7, UBUNTU1804, SUSE15
**Platforms**: CENTOS7,UBUNTU1804,SUSE15
**Profile**: regression

**Test Summary**: total: 5382, fail: 19, error: 0, timedout: 1, skip: 0, pass: 5362

### CODE REVIEW

File: test/fw/ptl/lib/pbs_testlib.py [hunk 1]
```diff
@@ -14647,6 +14647,11 @@ def __init__(self, job, cmd, host):
         self.cmd = cmd
         self.jobid = None
         self.hostname = host
+        self.runas_user = ""
```
Comment (hirenvadalia) [inline test/fw/ptl/lib/pbs_testlib.py:14650]: I think you can get rid of this instance variable, and use local scope variable in below `if shasta` block ([here](https://github.com/openpbs/openpbs/pull/1950/files#diff-fef109b45f82b3a763d1bc74763fb067R14704))

File: test/fw/ptl/lib/pbs_testlib.py [hunk 2]
```diff
@@ -14711,11 +14731,22 @@ def run(self):
                 _p.expect(out)
             self.logger.info('sending exit')
             _p.sendline("exit")
-            self.logger.info('waiting for the subprocess to finish')
-            _p.wait()
-            _p.close()
-            self.job.interactive_handle = None
-            self.logger.debug(_p.exitstatus)
+
+            if is_local:
```
Comment (hirenvadalia) [inline test/fw/ptl/lib/pbs_testlib.py:14735]: Not sure but do we really need this if else? I think we can use else part for both local/remote

File: test/fw/ptl/lib/pbs_testlib.py [hunk 3]
```diff
@@ -14647,6 +14647,11 @@ def __init__(self, job, cmd, host):
         self.cmd = cmd
         self.jobid = None
         self.hostname = host
+        self.runas_user = ""
+        if self.du.get_platform() == "shasta":
+            self.runas_user = PbsUser.get_user(job.username)
+            self.hostname = self.runas_user.host \
+                if self.runas_user.host else self.hostname
```
Comment (prakashcv13) [inline test/fw/ptl/lib/pbs_testlib.py:14654]: would it be simpler if we just have -

if self.runas_user.host

    self.hostname = self.runas_user.host

File: test/fw/ptl/lib/pbs_testlib.py (hunk 1 above)
Comment (prakashcv13) [inline test/fw/ptl/lib/pbs_testlib.py:14650]: set client = "" here
and assign it a value in the if shasta block.

we will not need another if shasta below.

File: test/fw/ptl/lib/pbs_testlib.py [hunk 4]
```diff
@@ -14711,11 +14731,22 @@ def run(self):
...
-            _p.close()
-            self.job.interactive_handle = None
-            self.logger.debug(_p.exitstatus)
+
+            if is_local:
+                self.logger.info('waiting for the subprocess to finish')
+                _p.wait()
+                _p.close()
+                self.job.interactive_handle = None
+                self.logger.debug(_p.exitstatus)
```
Comment (prakashcv13) [inline test/fw/ptl/lib/pbs_testlib.py:14740]: why debug?

File: test/fw/ptl/lib/pbs_testlib.py (hunk 2 above)
Comment (sujatapatnaik52) [reply test/fw/ptl/lib/pbs_testlib.py:14735]: I am not sure if we can combine both of them into one block. I will give it a try.

File: test/fw/ptl/lib/pbs_testlib.py (hunk 4 above)
Comment (sujatapatnaik52) [reply test/fw/ptl/lib/pbs_testlib.py:14740]: These are not really my changes. I would guess this is just a debug message to find out if the interactive shell has terminated gracefully.

**Comment by sujatapatnaik52** [comment]
[suspend_resume_accounting.txt](https://github.com/openpbs/openpbs/files/5034056/suspend_resume_accounting.txt)
[suspend_resume_accounting_remote.txt](https://github.com/openpbs/openpbs/files/5034057/suspend_resume_accounting_remote.txt)
[smoketest_logs.txt](https://github.com/openpbs/openpbs/files/5034059/smoketest_logs.txt)
[smoketest_remote_logs.txt](https://github.com/openpbs/openpbs/files/5034060/smoketest_remote_logs.txt)

**Comment by sujatapatnaik52** [comment]
License check was in pending state for my earlier commit hence force-pushed.

**Comment by sujatapatnaik52** [comment]
**Description:** Tests from given sources on platforms CENTOS7, UBUNTU1804, SUSE15
**Platforms:** CENTOS7,UBUNTU1804,SUSE15
**Profile:** regression

**Test Summary:** total: 5364, fail: 6, error: 0, timedout: 0, skip: 0, pass: 5358

**Pull request merged**